src/libcore/iter.rs

```diff
@@ -1400,11 +1400,11 @@ pub struct Chain<A, B> {
 }
 
 #[stable(feature = "rust1", since = "1.0.0")]
-impl<T, A, B> Iterator for Chain<A, B> where A: Iterator<Item=T>, B: Iterator<Item=T> {
-    type Item = T;
+impl<A, B> Iterator for Chain<A, B> where A: Iterator, B: Iterator<Item = A::Item> {
```

Comment:
I believe that's largely only used if the clause gets too long to fit on one line.